### PR TITLE
[BUGFIX release] Adds toJSON to list of descriptorTrap assertion exception

### LINKED ISSUE #16323: Ember 3.0 - Failed assertion when JSON.stringify -ing an Ember Object
Upgrading to Ember 3.0 and using ```JSON.stringify```, results in following failed assertion

```Assertion Failed: You attempted to access `someValue.toJSON` (on `<(unknown mixin):ember2486>`), but `someValue` is a computed property.↵↵Due to certain internal implementation details of Ember, ```

I read the linked [RFCs and discussion](https://github.com/emberjs/rfcs/pull/281) and I understand this is just a precautionary assertion, but this breaks ```JSON.stringify``` usage. 

https://github.com/douglascrockford/JSON-js/blob/master/json2.js#L237-L238

screen shot 2018-03-05 at 6 48 09 pm

### PR DESCRIPTION
**Pull request by mani-mishra**
Closes #16323

Changes as per suggestion in the linked ticket.

@rwjblue  @chancancode 👀 please

### CODE REVIEW

**Review by rwjblue — CHANGES_REQUESTED** [review]
Changes look great! Can you update the PR title and commit message to be `[BUGFIX release]` (that will help us make sure to get it pulled down into a 3.0.x)?

**Comment by mani-mishra** [comment]
Thanks, Updated.

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
Thank you!

File: packages/ember-metal/lib/properties.js [hunk 1]
```diff
@@ -148,6 +149,7 @@ if (EMBER_METAL_ES5_GETTERS) {
       });
 
       trap.toString = trap.valueOf = () => '[COMPUTED PROPERTY]';
+      trap.toJSON = trap.valueOf = () => '[COMPUTED PROPERTY]';
```
Comment (chancancode) [inline packages/ember-metal/lib/properties.js]: This should be folded into the previous line

Comment (mani-mishra) [reply packages/ember-metal/lib/properties.js]: Sorry. I did't understand the above statement.
Do you mean something like below?
```
trap.toString = trap.toSJON = trap.valueOf = () => '[COMPUTED PROPERTY]';
```
I can raise a PR, if needed.

Comment (rwjblue) [reply packages/ember-metal/lib/properties.js]: Yes, that’s what Godfrey was suggesting

Comment (mani-mishra) [reply packages/ember-metal/lib/properties.js]: Raised a PR for this #16365

**Pull request merged**